nestkernel/input_device.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *  input_device.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef INPUT_DEVICE_H
+#define INPUT_DEVICE_H
+
+// C++ includes:
+#include <fstream>
+#include <vector>
+
+// Includes from libnestutil:
+#include "lockptr.h"
+
+// Includes from nestkernel:
+#include "node.h"
+#include "device.h"
+#include "device_node.h"
+#include "input_backend.h"
+#include "nest_types.h"
+#include "kernel_manager.h"
+
+// Includes from sli:
+#include "dictdatum.h"
+#include "dictutils.h"
+
+namespace nest
+{
+
+class InputDevice : public DeviceNode, public Device
+{
+public:
+  InputDevice();
+  InputDevice( const InputDevice& );
+
+  void calibrate( const std::vector< Name >&, const std::vector< Name >& );
```

Comment:
This hides overloaded virtual functions. You can add `using` directives to avoid hiding overloaded virtual functions:
```c++
using Device::calibrate;
```
See also #1377.

Reply:
done